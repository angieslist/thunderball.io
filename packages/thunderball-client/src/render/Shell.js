import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { Router, RouterContext } from 'react-router';
import { IntlProvider } from 'react-intl';

// Shell of the body of the document
// This is called by client-side and server-side
class Shell extends React.Component {
  componentDidMount() {
    // Injection point: afterPageMounted
    if (this.props.injectors) {
      this.props.injectors
        .filter(injector => injector.afterPageMounted)
        .forEach(injector => injector.afterPageMounted(this.props.store, this.props.pageProps));
    }
  }

  render() {
    const { store, defaultLocale = 'en' } = this.props;

    return (
      <Provider store={store}>
        <IntlProvider
          defaultLocale={defaultLocale}
          initialNow={new Date()}
          key={defaultLocale} // https://github.com/yahoo/react-intl/issues/234
          locale={defaultLocale}
        >
          {
            process.env.IS_BROWSER ?
              <Router history={this.props.history}>
                {this.props.routes}
              </Router>
              : <RouterContext {...this.props.renderProps} />
          }
        </IntlProvider>
      </Provider>
    );
  }
}

Shell.propTypes = {
  store: PropTypes.object,
  pageProps: PropTypes.object,
  defaultLocale: PropTypes.string,

  // client side render
  injectors: PropTypes.array,
  history: PropTypes.object,
  routes: PropTypes.object,

  // server side render
  renderProps: PropTypes.object,
};

export default Shell;
