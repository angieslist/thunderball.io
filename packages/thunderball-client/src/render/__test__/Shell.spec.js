import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Shell from '../Shell';

jest.mock('react-redux');

const store = Object.freeze({
  dispatch: jest.fn(),
  getState: jest.fn(),
  testStore: true,
});
const pageProps = {
  testPageProps: true,
};
const injector = {
  afterPageMounted: jest.fn(),
};
const injectors = [injector];
const history = {
  testHistory: true,
};
const routes = {
  testRoutes: true,
};
const renderProps = {
  testRenderProps: true,
};

describe('Shell', () => {
  let app;
  beforeEach(() => {
    app = document.createElement('div');
    app.setAttribute('id', 'app');
    document.getElementsByTagName('body').item(0).appendChild(app);
  });
  it('renders ', () => {
    const DATE_TO_USE = new Date('2016');
    global.Date = jest.fn(() => DATE_TO_USE);
    const wrapper = shallow(
      <Shell
        store={store}
        pageProps={pageProps}
        injectors={injectors}
        history={history}
        routes={routes}
        renderProps={renderProps}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders in browser', () => {
    process.env.IS_BROWSER = true;
    const DATE_TO_USE = new Date('2016');
    global.Date = jest.fn(() => DATE_TO_USE);
    const wrapper = shallow(
      <Shell
        store={store}
        pageProps={pageProps}
        injectors={injectors}
        history={history}
        routes={routes}
        renderProps={renderProps}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    delete process.env.IS_BROWSER;
  });
  it('calls injector afterPageMounted', () => {
    shallow(
      <Shell
        store={store}
        pageProps={pageProps}
        injectors={injectors}
        history={history}
        routes={routes}
        renderProps={renderProps}
      />,
    );
    expect(injector.afterPageMounted).toHaveBeenCalledWith(store, pageProps);
  });
});
