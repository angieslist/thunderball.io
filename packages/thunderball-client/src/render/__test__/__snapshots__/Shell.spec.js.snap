// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shell renders  1`] = `
<Provider
  context={Object {}}
>
  <Provider
    store={
      Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction],
        "testStore": true,
      }
    }
  >
    <IntlProvider
      defaultLocale="en"
      initialNow={2016-01-01T00:00:00.000Z}
      key="en"
      locale="en"
    >
      <RouterContext
        createElement={[Function]}
        testRenderProps={true}
      />
    </IntlProvider>
  </Provider>
</Provider>
`;

exports[`Shell renders in browser 1`] = `
<Provider
  context={Object {}}
>
  <Provider
    store={
      Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction],
        "testStore": true,
      }
    }
  >
    <IntlProvider
      defaultLocale="en"
      initialNow={2016-01-01T00:00:00.000Z}
      key="en"
      locale="en"
    >
      <Router
        history={
          Object {
            "testHistory": true,
          }
        }
        render={[Function]}
      >
        <Component />
      </Router>
    </IntlProvider>
  </Provider>
</Provider>
`;
